### PR TITLE
Call api method before the external library is loaded

### PR DESCRIPTION
**Pull request by mrbatista**
Before call api method is necessary to ensure that external library is loaded.

``` javascript
angular.forEach(apiMethods, function(method) {
 ZendeskWidgetApi.prototype[method] = function() {
  $window.zE[method].apply($window.zE, arguments);
 };
});
```

It's possible that `$windows.zE[method]` is `undefined` if library is not loaded.
This PR fix this issue

### CODE REVIEW

**Comment by mrbatista** [comment]
@bsouthga PTAL

**Review by yangchristian — CHANGES_REQUESTED** [review]
Unfortunately [Travis is failing](https://travis-ci.org/CrossLead/angular-zendesk-widget/builds/160177575). Looks like there are loading problems with this approach to bootstrapping.

**Comment by mrbatista** [comment]
@yangchristian all api support "before page load". see [official documentation](https://developer.zendesk.com/embeddables/docs/widget/api)

**Comment by yangchristian** [comment]
Hmm, looks like then we'll need to adjust the tests so that continuous integration doesn't fail. In particular, we'll need to modify the [`mock service`](https://github.com/CrossLead/angular-zendesk-widget/blob/f6d271e030a622777b32420fdb720589c49c819b/test/unit/services/zendeskWidgetProviderSpec.js#L15). Let me know if you have time to take a look, otherwise I can probably get to it this weekend. Thanks!

File: src/angular-zendesk-widget/zendeskWidget.service.js [hunk 1]
```diff
@@ -42,7 +42,9 @@
 
           angular.forEach(apiMethods, function(method) {
             ZendeskWidgetApi.prototype[method] = function() {
-              $window.zE[method].apply($window.zE, arguments);
+              $window.zE(function() {
+                $window.zE[method].apply($window.zE, arguments);
```
Comment (yangchristian) [inline src/angular-zendesk-widget/zendeskWidget.service.js]: This is a bug: `arguments` is now blank due to the closure. Think you need to do something like this:

``` js
            ZendeskWidgetApi.prototype[method] = function() {
              var closureArgs = arguments; // Need to capture outer closure args
              $window.zE(function() {
                // arguments here is empty due to the new closure
                $window.zE[method].apply($window.zE, closureArgs);
              })
            };
```

**Comment by yangchristian** [comment]
@mrbatista: Couple updates:
- I updated the tests. Can you merge latest `master` into your branch?
- There is a bug with the change: please see the line note

Thanks!

File: src/angular-zendesk-widget/zendeskWidget.service.js (hunk 1 above)
Comment (mrbatista) [reply src/angular-zendesk-widget/zendeskWidget.service.js]: Fixed!

**Comment by mrbatista** [comment]
@yangchristian PTAL

**Review by yangchristian — APPROVED** [review]

**Comment by yangchristian** [comment]
@mrbatista: merged. Thanks for the contrib!

**Comment by mrbatista** [comment]
@yangchristian please publish new version on bower. Thanks!

**Comment by yangchristian** [comment]
@mrbatista: v0.1.1 up. Thanks again!

**Pull request merged**